Sources/Build/llbuild.swift

```diff
@@ -13,6 +13,21 @@ import Utility
 import PackageModel
 import PackageGraph
 
+/// An enum representing the different build options.
+public enum BuildOption {
+    /// Represents the option to build all targets and products (excluding test targets).
+    case allExcludingTests
+
+    /// Represents the option to build all targets and products (including test targets).
+    case allIncludingTests
+
+    /// Represents the option to build a specific target.
+    case target(ResolvedTarget)
+
+    /// Represents the option to build a specific product.
+    case product(ResolvedProduct)
```

Comment:
Right now, this enum is in `llbuild.swift` and the fact that the associated types are `ResolvedTarget` and `ResolvedPrduct` makes it unambiguous what they are referencing. If we make them names, it can be potentially ambiguous if they expect target/product names or LLBuild target names. The latter would be a bad idea IMHO because the LLBuild target names should stay private implementation detail of `LLBuildManifestGenerator`. On the other hand, if they are target/product names, it means doesn't it makes it harder to make sure we translate the name to a `ResolvedTarget/ResolvedProduct` at only one place in the code? For example, in my next PR, I will change the way the `ResolvedTarget/ResolvedProduct` are generated for this option so it also goes and look for targets/products which are not depended on.